spec/requests/api/provision_requests_spec.rb

```diff
@@ -105,6 +105,140 @@ def app
+    end
+    let(:az)             { FactoryGirl.create(:availability_zone_amazon) }
+    let(:cloud_network1) { FactoryGirl.create(:cloud_network, :ems_id => ems.id, :enabled => true) }
+    let(:cloud_subnet1)  { FactoryGirl.create(:cloud_subnet, :cloud_network_id => cloud_network1.id) }
+
+    let(:provreq_body) do
+      {
+        "template_fields" => {"guid" => template.guid},
+        "requester"       => {"user_name" => api_config(:user)}
+      }
+    end
+
+    let(:provreq1_body) do
+      provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_auto"              => false,
+          "placement_availability_zone" => az.id,
+          "cloud_network"               => cloud_network1.id,
+          "cloud_subnet"                => cloud_subnet1.id,
+          "security_groups"             => 1,
+          "floating_ip_address"         => 1
+        }
+      )
+    end
+
+    let(:provreq2_body) do
+      provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_availability_zone" => az.id
+        }
+      )
+    end
+
+    let(:provreq3_body) do
+      provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_auto"              => true,
+          "placement_availability_zone" => az.id
+        }
+      )
+    end
+
+    let(:expected_provreq_attributes) { %w(id options) }
+
+    let(:expected_provreq_hash) do
+      {
+        "userid"         => api_config(:user),
+        "requester_name" => api_config(:user_name),
+        "approval_state" => "pending_approval",
+        "type"           => "MiqProvisionRequest",
+        "request_type"   => "template",
+        "message"        => /Provisioning/i,
+        "status"         => "Ok"
+      }
+    end
+
+    it "supports manual placement" do
+      api_basic_authorize collection_action_identifier(:provision_requests, :create)
+
+      # dialog  # Create the Provisioning dialog
+      aws_dialog # Create the AWS Provisioning dialog
+      run_post(provision_requests_url, provreq1_body)
+
+      # pp @result
+      expect_request_success
+      expect_result_resources_to_include_keys("results", expected_provreq_attributes)
+      expect_results_to_match_hash("results", [expected_provreq_hash])
+
+      options = @result["results"].first["options"]
+      # pp options
+      expect(options["placement_auto"]).to eq([false, 0])
+      expect(options["placement_availability_zone"]).to eq(['xxx'])
+      expect(options["cloud_network"]).to eq(['xxx'])
+      expect(options["cloud_subnet"]).to eq(['xxx'])
+      expect(options["security_groups"]).to eq(['xxx'])
+      expect(options["floating_ip_address"]).to eq(['xxx'])
```

Comment:
The options key for most of the drop-down fields end up as an array that stores [value, display text].  A field like `cloud_subnet` ends up with a generated display_name that can include the AZ name and/or the `cidr` value.

For your purposes I think you just want to validate that the proper value (ID of the resource) is stored.

```ruby
expect(options["placement_availability_zone"].first).to eq(az.id)
expect(options["cloud_network"].first).to eq(cloud_network1.id)
expect(options["cloud_subnet"].first).to eq(cloud_subnet1.id)
expect(options["security_groups"].first).to be nil
expect(options["floating_ip_address"].first).to be nil
```